### PR TITLE
Introduce isReadyForUser and verify in file transfer ownership 

### LINKED ISSUE #23786: files:transfer-ownership command does not work in some scenarios with encryption
### Steps to reproduce

1. Create 2 oC users, user1 and user2
2. Log in with user1
3. Create a file.txt
4. Log in as admin user and encrypt the server
5. Log out and log in as admin.
6. As admin via CLI run  -->  sudo -u www-data ./occ files:transfer-ownership user1 user2

### Expected behaviour

The ownership of files should be transferred from user1 to user2 

### Actual behaviour

An error is shown and some of the files are not transferred

### Server configuration
**Operating system**:
Ubuntu 14.04

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
5.5.9

**ownCloud version:**"9.0.1.2","versionstring":"9.0.1 RC2","edition":"Enterprise"}

**Updated from an older ownCloud or fresh install:**
Fresh

**List of activated apps:**
Enabled:
  - activity: 2.2.1
  - admin_audit: 0.7
  - comments: 0.2
  - dav: 0.1.6
  - encryption
  - enterprise_key: 0.1.0
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firewall: 2.3.0
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
  - windows_network_drive: 0.2.33
  - workflow: 0.1.3
Disabled:
  - announcementcenter
  - external
  - files_antivirus
  - files_drop
  - files_ldap_home
  - objectstore
  - password_policy
  - sharepoint
  - user_external
  - user_ldap
  - user_shibboleth

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** 
No

**Logs**
```
{"reqId":"U8ld8vodFVW\/2RmfTb3D","remoteAddr":"","app":"PHP","message":"Class 'OC_Mount_Config' not found at \/opt\/owncloud\/apps\/files_external\/appinfo\/register_command.php#38","level":3,"time":"2016-04-04T16:15:31+00:00","method":"--","url":"--","user":"--"}
{"reqId":"JuExhCziPp6jbWEKxrC3","remoteAddr":"","app":"PHP","message":"fclose() expects parameter 1 to be resource, string given at \/opt\/owncloud\/lib\/private\/files\/storage\/wrapper\/encryption.php#748","level":3,"time":"2016-04-04T16:18:07+00:00","method":"--","url":"--","user":"--"}
```
### Client configuration

**browser**

 Firefox

screen shot 2016-04-04 at 18 19 21

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #23786

### CODE REVIEW

File: apps/files/command/transferownership.php [hunk 1]
```diff
@@ -97,6 +97,9 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 			$output->writeln("<error>Unknown destination user $this->destinationUser</error>");
 			return;
 		}
+		
+		// target user has to be ready
+		\OC::$server->getEncryptionManager()->isReadyForUser($this->destinationUser);
```
Comment (nickvergessen) [inline apps/files/command/transferownership.php]: You are not doing anything with the returned boolean?

**Comment by davitol** [comment]
Switching the patch and following https://github.com/owncloud/core/issues/23786 steps:

PHP Fatal error:  Call to a member function isReadyForUser() on a non-object in /opt/owncloud/lib/private/encryption/manager.php on line 129

**Comment by DeepDiver1975** [comment]
@davitol THX for testing - mind retesting? THX

**Comment by davitol** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
@schiesbn please review the encryption related changes - THX

**Comment by schiesbn** [comment]
code looks good! :+1: 

Thanks for picking this up, Thomas!

**Comment by DeepDiver1975** [comment]
@karlitschek we should backport this - #23786 is assigned to 9.0.2

Note: this holds a public api change - https://github.com/owncloud/core/pull/24004/files#diff-8d7e0ca7514556cd6929b4c073c9629dR179

Generally speaking we should not backport ths since it will immediately break all existing implementation of IEncryptionModule

**Comment by karlitschek** [comment]
ok. looks safe. please backport 👍

**Pull request merged**